### PR TITLE
Redesigned compat table macro

### PR DESCRIPTION
**Pull request by Elchi3**
To test this, you will need:
- Kuma with https://github.com/mozilla/kuma/pull/4436.
- A waffle flag called `compat_data` enabled.
- A test page with some compat data calls. (TBD, I'll post one here).

How this works:
- Compat.ejs is mostly untouched and additionally loads CompatBeta.ejs, but all HTML that is generated by CompatBeta is hidden.
- Only if the waffle flag is on, the CompatBeta output is shown, and the Compat output is hidden.
- This allows us to do A/B or user testing.
- Once we are done with testing, CompatBeta.ejs will be copied into Compat.ejs, and CompatBeta.ejs can be deleted. (On the pages, no macro calls need to be updated. Before, during and after the beta the {{compat("something.something")}} calls are the same).

Testing:
- run tests with `make test-macros` in the kumascript dir, it might require a `make build-kumascript` in the kuma dir before.
- The CompatBeta output is tested by test-compat.js and compat data fixtures.
- I've added basic tests, we should expand this going forward. There are many cases in compat data.

### CODE REVIEW

**Comment by stephaniehobson** [comment]
Here's my testing page code:

```
<h2 id="HTML">HTML</h2>

<p>{{compat("html.elements.link", 2)}}</p>

<h2 id="CSS">CSS</h2>

<div>{{Compat("css.properties.background-attachment")}}</div>

<div>{{Compat("css.properties.background")}}</div>

<h2 id="API">API</h2>

<div>{{Compat("api.WebGLRenderingContext.WebGLRenderingContext")}}</div>

<h2 id="JS">JS</h2>

<div>{{Compat("javascript.builtins.Promise.Promise")}}</div>

<h2 id="HTTP">HTTP</h2>

<div>{{{Compat("http.status.404")}}</div>

<p>{{Compat("http.methods.CONNECT")}}</p>

<h2 id="WebExtensions">WebExtensions</h2>

<p>{{compat("webextensions.api.alarms", 3)}} {{compat("webextensions.api.alarms.Alarm")}} {{WebExtBrowserCompat("webextensions.manifest")}} {{WebExtBrowserCompat("webextensions.applications")}}</p>

<h2 id="Test">Test</h2>

<p>{{compat("test.sample.api.featureWithSubfeatures")}}</p>
```

**Comment by stephaniehobson** [comment]
So many improvements I want to make to this seeing it with real data 0.0

**Comment by stephaniehobson** [comment]
Did a small amount of QA on the kuma branch. The buttons should look better and the weird notch on the top left border is gone.

File: macros/CompatBeta.ejs [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+    output += writeIcon(platformId);
+    output += '</th>';
+  }
+
+  output += '</tr>';
+  return output;
+}
+
+function writeCompatBrowsersRow() {
+  let output = '<tr>';
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply macros/CompatBeta.ejs]: Thanks, added :)

**Comment by Elchi3** [comment]
Added some more test fixtures and basic tests.
I also manually tested some more cases, and it seems to be working fine. Prod data would give me even more test cases, of course. I think this is ready but I'm not sure who will review this now....

Front-end wise I noticed:
- "Show old table" is not working for me properly, I would appreciate a double-check.
- There are missing icons: "server", "nodejs", "deprecated".

**Comment by stephaniehobson** [comment]
"Show old table" is working for me on pages with only one table. It doesn't work on my test page with more than one.

Could you please edit the macro to add `bc-table-js` `bc-table-ext` (or something like that) to the <table> tag for the javascript and web extensions tables respectively. I'll treat the third configuration as the default.

I can 👍 the functionality. I'll have a look at the macro code itself this afternoon.

**Comment by stephaniehobson** [comment]
Sorry, those should be `bc-table_js` and `bc-table_ext` and I need a third one after all, let's go with `bc-table_web`.

**Comment by Elchi3** [comment]
So, you dont need `bc-table-${browserCount}"`anymore?

**Comment by stephaniehobson** [comment]
Correct, you can remove `bc-table-${browserCount}`

File: macros/CompatBeta.ejs [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+
+/*
+Create notes section
+
+`supportData` is a support_statement
+`browserId` is a compat_block browser ID
+
+*/
+function writeNotes(support, browserId) {
+  let output = '<section class="bc-history" aria-hidden="true"><dl>';
```
Comment (stephaniehobson) [inline macros/CompatBeta.ejs]: Do you have any thoughts on changing this to an `<aside>`? Not sure I picked the right element here.

File: macros/CompatBeta.ejs [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+// see https://github.com/mdn/browser-compat-data/issues/546
+
+`supportData` is a support_statement
+`browserId` is a compat_block browser ID
+*/
+function writeFlagsNote(supportData, browserId) {
+  let output = '';
+
+  const firefoxPrefs = 'To change preferences in Firefox, visit about:config.';
+  const chromePrefs = 'To change preferences in Chrome, visit chrome://flags.';
```
Comment (stephaniehobson) [inline macros/CompatBeta.ejs]: We're localizing some strings but not most of the strings in `writeFlagsNote`. They look hard though. Not a blocker for the beta but something that should be on a to-do list somewhere.

File: macros/CompatBeta.ejs [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+
+function writeCompatPlatformsRow() {
+  let output = '<tr class="bc-platforms">';
+  output += '<td></td>';
+
+  let platforms = ['desktop', 'mobile'];
+  if (category === 'javascript') {
+    platforms.push('server');
+  }
+  if (category === 'webextensions') {
```
Comment (stephaniehobson) [inline macros/CompatBeta.ejs]: Can this switching on table type stuff be moved up to line 60ish where the other switching is done?

File: macros/L10n-CompatTable.json [hunk 5]
```diff
@@ -49,12 +72,12 @@
...
-        "es"   : "(Si)",
-        "fr"   : "(Oui)",
-        "ja"   : "(有り)",
-        "ru"   : "(Да)"
+        "en-US": "Yes",
+        "de"   : "Ja",
+        "es"   : "Si",
+        "fr"   : "Oui",
+        "ja"   : "有り",
+        "ru"   : "Да"
```
Comment (stephaniehobson) [inline macros/L10n-CompatTable.json]: 🎉 already having these localized!

File: macros/L10n-CompatTable.json [hunk 6]
```diff
@@ -123,151 +146,132 @@
...
+    "bc_icon_title_altname": {
+        "en-US": "Uses the non-standard name: <code>$1$</code>",
+        "de"   : "Verwendet den nicht standardisierten Namen: <code>$1$</code>",
+        "es"   : "Usa un nombre no estandar : <code>$1$</code> ",
+        "fr"   : "Cette fonctionnalité utilise le nom non-standard&nbsp;: <code>$1$</code>",
+        "ja"   : "非標準の名前を使用: <code>$1$</code>",
+        "ru"   : "Использует нестандартное имя: <code>$1$</code>"
     },
-    "requirement_protected": {
+    "bc_icon_title_protected": {
```
Comment (stephaniehobson) [inline macros/L10n-CompatTable.json]: LOL. We have these in the macro before the schema 😝

File: macros/CompatBeta.ejs (hunk 2 above)
Comment (Elchi3) [reply macros/CompatBeta.ejs]: `<aside>` might be better, but I'm really not an HTML or semantics expert :confused:

File: macros/CompatBeta.ejs [hunk 7]
```diff
@@ -0,0 +1,603 @@
...
+  let output = '';
+  switch (added) {
+    case null:
+      output = `<abbr title="${localize(compatStrings, 'supportsShort_unknown_title')}">
+                ${localize(compatStrings, 'supportsShort_unknown')}
+              </abbr>`;
+    break;
+    case true:
+      output = `<abbr title="${localize(compatStrings, 'supportsLong_yes')}"
+                class="bc-level bc-level-yes only-icon">
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply macros/CompatBeta.ejs]: I removed "bc-level"

File: macros/CompatBeta.ejs (hunk 3 above)
Comment (Elchi3) [reply macros/CompatBeta.ejs]: Added a TODO

File: macros/CompatBeta.ejs [hunk 8]
```diff
@@ -0,0 +1,603 @@
...
+    output += ':';
+    flagTextStart = ' this';
+  }
+
+  let flagText = `${flagTextStart} feature is behind the <code>${supportData.flag.name}</code>`;
+
+  // value_to_set is optional
+  let valueToSet = '';
+  if (supportData.flag.value_to_set) {
+    valueToSet = ` (needs to be set to <code>${supportData.flag.value_to_set}</code>)`;
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply macros/CompatBeta.ejs]: Good point. Tweaked.

File: macros/CompatBeta.ejs (hunk 4 above)
Comment (Elchi3) [reply macros/CompatBeta.ejs]: Yes, good call. Changed that.

**Pull request merged**